examples/client.c

```diff
@@ -12,18 +12,37 @@
 
 #include <string.h>
 #include <stdlib.h>
-#include <unistd.h>
+#ifdef HAVE_UNISTD_H
+# include <unistd.h>
+#else
+# include "getopt.h"
+#endif
 #include <stdio.h>
 #include <ctype.h>
-#include <sys/select.h>
+#ifdef HAVE_SYS_SELECT_H
+# include <sys/select.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/socket.h>
-#include <netinet/in.h>
-#include <arpa/inet.h>
-#include <netdb.h>
+#ifdef HAVE_SYS_SOCKET_H
+# include <sys/socket.h>
+#endif
+#ifdef HAVE_NETINET_IN_H
+# include <netinet/in.h>
+#endif
+#ifdef HAVE_ARPA_INET_H
+# include <arpa/inet.h>
+#endif
+#ifdef HAVE_NETDB_H
+# include <netdb.h>
+#endif
+#ifndef HAVE_STRCASECMP
+  /* The POSIX strcasecmp() is not supported, so use the ISO C++ conformant _stricmp instead. */
+# define strcasecmp _stricmp
```

Comment:
I wonder where _stricmp() might be defined. In my (draft version) of ISO 14882:2011, there is no such thing. Please enlighten me.